py/emitnative.c

```diff
@@ -2154,7 +2154,7 @@ STATIC void emit_native_with_cleanup(emit_t *emit, mp_uint_t label) {
     ASM_MOV_REG_LOCAL(emit->as, REG_ARG_1, LOCAL_IDX_EXC_VAL(emit)); // get exc
 
     // Check if exc is None and jump to non-exc handler if it is
-    emit_native_mov_reg_const(emit, REG_ARG_2, MP_F_CONST_NONE_OBJ);
+    ASM_MOV_REG_IMM(emit->as, REG_ARG_2, (mp_uint_t)MP_OBJ_NULL);
     ASM_JUMP_IF_REG_EQ(emit->as, REG_ARG_1, REG_ARG_2, *emit->label_slot + 2);
```

Comment:
I think these 2 lines can be optimised into 1 line using `ASM_JUMP_IF_REG_ZERO`.  We assume that `MP_OBJ_NULL` is zero.

Reply:
Done. I set `bool_test` to `false` which I think is correct here (i.e. it's a strict non-zero test).

Reply:
correct